### PR TITLE
Adding SAI PTF scripts for Extended testcases

### PR DESCRIPTION
**Pull request by chitra-raghavan**
Adding below tests into SAI PTF tests (https://github.com/opencomputeproject/SAI/blob/master/test/saithrift/tests/SAITest.md)

L2MtuTest
L3MtuTest
L2MacMoveTestI 
L2MacMoveTestII
L2MacMoveTestIII
L3IPv4NeighborMacTest
L3IPv6NeighborMacTest

Added sai_thrift_set_neighbor_attribute proc in switch.py
Test scripts  are validated with SAI version 1.3.3

### CODE REVIEW

File: test/saithrift/src/switch_sai_rpc_server.cpp [hunk 1]
```diff
@@ -400,6 +403,9 @@ class switch_sai_rpcHandler : virtual public switch_sai_rpcIf {
                   break;
               case SAI_ROUTER_INTERFACE_ATTR_INGRESS_ACL:
                   attr_list[i].value.oid = attribute.value.oid;
```
Comment (lguohan) [inline test/saithrift/src/switch_sai_rpc_server.cpp]: add break;

File: test/saithrift/tests/sail2.py [hunk 2]
```diff
@@ -757,3 +757,570 @@ def runTest(self):
...
+                                ip_dst='192.168.0.1',
+                                ip_src='192.168.0.2',
+                                ip_id=105,
+                                ip_ttl=64)
+        try:
+
+            ### Sending packet from Port1
+            send_packet(self, 0, str(pkt1))
+
+            ### verify the packet @ Port2
```
Comment (lguohan) [inline test/saithrift/tests/sail2.py]: verify the packet @ Port2 and Port3

File: test/saithrift/tests/sail2.py [hunk 3]
```diff
@@ -757,3 +757,570 @@ def runTest(self):
...
+
+            ### verify the packet @ Port1
+            verify_packets(self, exp_pkt2, [0])
+
+            time.sleep(1)
+
+            ### Sending packet from Port1
+            send_packet(self, 0, str(pkt1))
+
+            ### verify the packet @ Port3
```
Comment (lguohan) [inline test/saithrift/tests/sail2.py]: verify the packet @ Port2

File: test/saithrift/tests/sail2.py [hunk 4]
```diff
@@ -757,3 +757,570 @@ def runTest(self):
...
+            verify_packets(self, exp_pkt1, [1])
+
+            ### verify the packet @ Port3
```
Comment (lguohan) [inline test/saithrift/tests/sail2.py]: verify no packet @ Port 3

File: test/saithrift/tests/sail2.py [hunk 5]
```diff
@@ -757,3 +757,570 @@ def runTest(self):
...
+
+            ### verify the packet @ Port1
+            verify_packets(self, exp_pkt2, [0])
+
+            time.sleep(1)
+
+            ### Send packet (src_mac=MAC1 and dst_mac=MAC2) from port 1
+            send_packet(self, 0, str(pkt1))
+
+            ### verify the packet @ Port3 & Port2
```
Comment (lguohan) [inline test/saithrift/tests/sail2.py]: verify the packet @Port 3 and no packet at @Port2

File: test/saithrift/tests/sail2.py [hunk 6]
```diff
@@ -757,3 +757,570 @@ def runTest(self):
...
+        forwarding. Execute "L2MacMoveTest" and let MAC1 be learnt on 2nd port. Verify L3
+        forwarding and ensure that traffic is now forwarded to the newly learnt port.
+
+        Steps:
+        1. Create VLAN 10 and associate 2 untagged ports (port 1,2) as the member port of VLAN 10.
+        2. Set attribute value of "Port VLAN ID 10" for the ports 1, 2
+        3. Create Virtual router V1 and enable V4.
+        4. Create  virtual router interfaces and set the interface type as VLAN for VLAN 10
+           and create one more port3 (RIF Id2).
+        5. Create IPv6 neighbor entry with MAC1 and asociate with "RIF Id1".
```
Comment (lguohan) [inline test/saithrift/tests/sail2.py]: Looks like a typo, it should be IPv4 neighbor entry.

File: test/saithrift/tests/sail2.py [hunk 7]
```diff
@@ -757,3 +757,570 @@ def runTest(self):
...
+        ip_addr1_subnet = '192.168.0.0'
+        ip_mask1 = '255.255.255.0'
+
+        ### create vlan
+        vlan_oid = sai_thrift_create_vlan(self.client, vlan_id)
+
+        ### create vlan membership with ports
+        vlan_member1 = sai_thrift_create_vlan_member(self.client, vlan_oid, port1, SAI_VLAN_TAGGING_MODE_UNTAGGED)
+        vlan_member2 = sai_thrift_create_vlan_member(self.client, vlan_oid, port2, SAI_VLAN_TAGGING_MODE_UNTAGGED)
+        vlan_member3 = sai_thrift_create_vlan_member(self.client, vlan_oid, port3, SAI_VLAN_TAGGING_MODE_UNTAGGED)
```
Comment (lguohan) [inline test/saithrift/tests/sail2.py]: according to above description, port 3 is not in the vlan 10 but is a port-based router interface. can you correct that?

File: test/saithrift/tests/sail2.py (hunk 3 above)
Comment (chitra-raghavan) [reply test/saithrift/tests/sail2.py]: corrected description

File: test/saithrift/tests/sail2.py (hunk 5 above)
Comment (chitra-raghavan) [reply test/saithrift/tests/sail2.py]: corrected description

File: test/saithrift/tests/sail2.py (hunk 7 above)
Comment (chitra-raghavan) [reply test/saithrift/tests/sail2.py]: corrected description,  in accordance with script

File: test/saithrift/tests/sail3.py [hunk 8]
```diff
@@ -2025,3 +2025,431 @@ def runTest(self):
...
+        Steps:
+        1. Create VLAN 100, 200 in the database
+        2. Associate 2 tagged ports (port 1, 2) as the member port of VLAN 100 and 200 respectively.
+        3. Create Virtual router V1 and enable V4.
+        4. Create two virtual router interfaces and set the interface type as VLAN
+        5. Create IPv4 neighbor entry with MAC1 and asociate with ""RIF Id1"".
+        6. Create next hop, route to reach the neighbor (MAC1).
+        7. Send test packet from port 1 with src_mac = MAC1 address and dst_mac = router MAC.
+        8. Send traffic on port 2 and observe traffic forwarding to port 1.
+        9. Change the neighbor attribute to update with new MAC (MAC3)
```
Comment (atitjain) [inline test/saithrift/tests/sail3.py]: You can add a step here to send the packet with MAC1 still and see the behavior before learning FDB entry for MAC3.

**Pull request merged**